common/db/db.go

```diff
@@ -66,18 +69,57 @@ func CloseMongoConn() {
 func checkDBConnection() {
 	errorCounter := 0
 	for {
-		if errorCounter < 10 {
-			time.Sleep(1 * time.Second)
-			err := mongoClient.Ping(mongoClientCtx, nil)
-			if err != nil {
-				fmt.Println("FAILED PINGING MONGO")
-				errorCounter++
-				continue
+		time.Sleep(1 * time.Second)
+		err := mongoClient.Ping(mongoClientCtx, nil)
+
+		if err != nil {
+			errorCounter++
+			log.WithFields(log.Fields{
+				"error_count": errorCounter,
+				"error":       err,
+			}).Warn("Failed to ping MongoDB")
+
+			if err := reconnectMongo(); err != nil {
+				log.WithFields(log.Fields{
+					"error_count": errorCounter,
+					"error":       err,
+				}).Error("Failed to reconnect to MongoDB")
+
+				if errorCounter >= 10 {
```

Comment:
Would you mind increasing this to 30? I've been thinking for a while that this is too strict, you should also change the log message below along with it